big_tests/tests/presence_SUITE.erl

```diff
@@ -236,19 +238,18 @@ invisible_presence(Config) ->
         escalus:send(Bob, escalus_stanza:roster_add_contact(Alice,
                                                             [<<"enemies">>],
                                                              <<"Alice">>)),
-        PushReqB = escalus:wait_for_stanza(Bob),
-        escalus:assert(is_roster_set, PushReqB),
+        Response1 = escalus:wait_for_stanzas(Bob, 2),
+        escalus:assert_many([is_roster_set, is_iq_result], Response1),
+        [PushReqB] = lists:filter(fun(S) -> escalus_pred:is_roster_set(S) end, Response1),
         escalus:send(Bob, escalus_stanza:iq_result(PushReqB)),
-        escalus:assert(is_iq_result, escalus:wait_for_stanza(Bob)),
 
         %% Bob sends subscribed presence
         escalus:send(Bob, escalus_stanza:presence_direct(AliceJid, <<"subscribed">>)),
 
         %% Alice receives subscribed
-        Stanzas = escalus:wait_for_stanzas(Alice, 2),
+        Stanzas = escalus:wait_for_stanzas(Alice, 3),
 
-        check_subscription_stanzas(Stanzas, <<"subscribed">>),
-        escalus:assert(is_presence, escalus:wait_for_stanza(Alice)),
+        check_subscription_stanzas(Stanzas, <<"subscribed">>, <<"available">>),
```

Comment:
I am wondering, how many clients relay on the fact that roster updates come before presences in order to correctly display it. Though there is no information about the order in XMPP RFC, so...

Reply:
Any client that relies on it is not conformant with specification, so we don't care :) The issue is that our tests often do, hence so many changes to test suites.

Reply:
@bartekgorny Does this pull request change the order in which roster updates and presences are sent? If so, why is that? Is the order still deterministic (I am assuming it was before the change)?

Reply:
Not explicitly, but somehow yes. Previously part of these messages was sent by the iq handling process, and part was delegated to the user's c2s processes. Currently everything is handled by c2s process, so it is less deterministic. The real issue is that our tests often do a series of wait_for_stanza+assert, instead of one assert_many.